### PR TITLE
New functions - tests

### PR DESCRIPTION
**Pull request by tomblind**
This PR adds a full suite of tests for function assignments (both valid and invalid).

It also addresses edge cases found, specifically a case where lambda-class-properties marked with 'this: void' would still be considered method-like.

### CODE REVIEW

File: src/Errors.ts [hunk 1]
```diff
@@ -71,17 +71,25 @@ export class TSTLErrors {
...
+                                      + `To fix, wrap the method in an arrow function.`,
+                                      node);
         }
     }
 
     public static UnsupportedMethodConversion = (node: ts.Node, name?: string) => {
         if (name) {
-            return new TranspileError(`Unsupported conversion from function to method "${name}".`, node);
+            return new TranspileError(`Unsupported conversion from function to method "${name}". `
+                                      + `To fix, wrap the function in an arrow function.`,
```
Comment (Perryvw) [inline src/Errors.ts]: Should be a very specific arrow function right?

File: test/unit/assignments.spec.ts [hunk 2]
```diff
@@ -6,6 +6,23 @@ const fs = require("fs");
 
 export class AssignmentTests {
```
Comment (Perryvw) [inline test/unit/assignments.spec.ts]: Does this mean we can get rid of the assignments translation test file? Or is it already gone on this branch?

File: src/Errors.ts (hunk 1 above)
Comment (tomblind) [reply src/Errors.ts]: The idea is that function expressions can have their context type deduced based on what they are being assigned to, so a simple arrow function should suffice. There's a few edge cases where this doesn't work and 'this' has to be explicitly stated or you will still get this error, though.

File: test/unit/assignments.spec.ts (hunk 2 above)
Comment (tomblind) [reply test/unit/assignments.spec.ts]: Not in this branch, but it's removed in master, so when things get merged it should be gone.
